src/Components/OATGraphViewer/Internal/OATGraphCustomEdge.tsx

```diff
@@ -57,6 +57,15 @@ const OATGraphCustomEdge: React.FC<IOATGraphCustomEdgeProps> = ({
         setNameText(getPropertyDisplayName(data));
     }, [data]);
 
+    useEffect(() => {
+        if (!state.model) {
+            onNameBlur();
+        }
+        if (state.model && state.model['@id'] !== data.id) {
+            onNameBlur();
+        }
+    }, [state]);
```

Comment:
Can't we combine these `if` statements with an `||`?

Aslo, can we make this effect dependent on model, rather than the whole state? I.e. extract model `const { model } = state;` and then use that as the dependency.